### PR TITLE
feat: update VMSS node pools

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR introduces the new CLI feature `aks-engine update --node-pool <node pool name>` to update VMSS node pools on existing cluster via AKS Engine ARM template generation.

tl;dr With a new version of AKS Engine compared to the version used to build your cluster, and/or with changes to the agentPoolProfiles definition in the generated `apimodel.json` output, you use `aks-engine update` to essentially push a new VMSS model to your node pool. Subsequent scale out operations against that VMSS will therefore use the updated VMSS model.

Some real-world examples:

- change the VM SKU
- change the OS distro
- change other AKS Engine-exposed OS configuration (e.g., sysctld)

In practice the implementation essentially re-uses `aks-engine scale`, giving that preëxisting command the right inputs to do an ARM template deployment against an existing VMSS with no change to the `Count` value.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
Worth mentioning: detailed documentation will follow this PR (as part of a larger set of improved documentation around all CLI functionality)

File: cmd/update.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+				if !sc.vmInAgentPool(vmssName, vmss.Tags) {
+					continue
+				}
+			}
+
+			if vmss.Sku != nil {
+				sc.newDesiredAgentCount = int(*vmss.Sku.Capacity)
+				uc.agentPool.Count = sc.newDesiredAgentCount
+			} else {
+				return errors.Wrap(err, fmt.Sprintf("failed to detect find VMSS matching node pool %s in resource group %s", sc.agentPoolToScale, sc.resourceGroupName))
```
Comment (mboersma) [inline cmd/update.go:249]:
```suggestion
				return errors.Wrap(err, fmt.Sprintf("failed to find VMSS matching node pool %s in resource group %s", sc.agentPoolToScale, sc.resourceGroupName))
```

**Review by mboersma — COMMENTED** [review]
This makes sense to me and should help users quite a bit.

We should update the `aks-engine help` example output in docs/community/developer-guide.md to include this command.

**Comment by jackfrancis** [comment]
> This makes sense to me and should help users quite a bit.
> 
> We should update the `aks-engine help` example output in docs/community/developer-guide.md to include this command.

Done

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**